pkg/operator/discover/discover.go

```diff
@@ -85,9 +87,17 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage, securityAcc
 		discovery_parameters = append(discovery_parameters, "--use-ceph-volume")
 	}
 
+	// Get the operator pod details
+	// So that we can attach the owner reference to the discover daemon set
+	operatorPod, err := k8sutil.GetRunningPod(d.clientset)
```

Comment:
If we set the owner reference to the pod, they will be deleted every time there is an upgrade or the operator pod is deleted to trigger a restart, etc. Instead of the pod, let's set the owner reference to the `rook-ceph-operator` deployment. It should be safer to use as the parent. It also has a consistent name so we should just be able to get it from K8s based on name and the same namespace.

Reply:
I tried that and it kept running. Currently, the op deployment has no owner ref as far as I saw.

Reply:
I think the owner reference of the pod is set to the replica set one, that's why it's still running.

Reply:
Yes, deleting the rs makes the discover disappear. I think it's acceptable, isn't it?

Reply:
Ah, I see now, it's actually using the owner of the pod, which wouldn't be disturbed during upgrade or other pod restarts. This makes sense.